packages/styled-components/components/fragment/index.js

```diff
@@ -12,22 +15,18 @@ const Fragment = (props) => {
     style,
     theme,
   } = props;
-
-  let {
-    tag,
-  } = props;
-
   const { Element } = theme;
-
-  // If we have something in `style` and element is empty, require a tag.
-  if (null === tag && Object.keys(style).length && ! style.length) {
-    tag = 'span';
-  }
+  const standardProps = useStandardProps(props);
+  const { as } = standardProps;
+  const useTag = (
+    as ||
+    (! as && Object.keys(style).length && ! style.length)
```

Comment:
The `! as` in the second grouping is unnecessary, since if it is a truthy value, this will never be evaluated.